_drafts/SC1-3-1-tables-headers-id-correct.md

```diff
@@ -0,0 +1,131 @@
+---
+rule_id: SC1-3-1-tables-headers-id-correct
+name: Tables headers-id associations correct
+test_mode: semi-automatic
+environment:  DOM Structure
+
+success_criterion:
+- 1.3.1 # Info and relationships (level A)
+
+authors:
+- Ken Petri
+---
+
+## Description
+
+This rule checks that data table headers-id associations reference correct targets in the correct order.
+
+## Background
+
+- [F90: Failure of Success Criterion 1.3.1 for incorrectly associating table headers and content via the headers and id attributes](https://www.w3.org/TR/WCAG20-TECHS/F90.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H43.html)
+- [F46: Failure of Success Criterion 1.3.1 due to using th elements, caption elements, or non-empty summary attributes in layout tables](https://www.w3.org/TR/WCAG20-TECHS/F46.html)
+
+## Assumptions
+
+- Tables are [well-formed, according to the HTML5.1 specification.](https://www.w3.org/TR/html51/tabular-data.html#forming-a-table).
+
+## Test procedure
+
+### Selector
+
+Loop outward from most deeply nested table.
+
+For each un-marked table, select all elements that match the following CSS selector:
+
+    table:not([role=presentation]) th[headers],
```

Comment:
The following table would match this, even though I don't think that it should:

```html
<table><tr><td>
   <table role="presentation"><tr><td headers="">Matching cell</td></tr></table>
</td></tr></table>
```

The same is true for the selector in the second rule

Reply:
@EmmaJP and @cpandhi I changed the rule so that it does a very basic check for the existence of the th[headers] or td[headers] and then just assumes that the author is using these within a data table. Thought this was the simplest approach.

Reply:
@kensgists Where can I find these changes? I want to make sure we've got a PR for this.